### DISCUSSION TITLE
MySQL SSL pem certificates

### DISCUSSION THREAD

**Discussion by gerickt** [FreshRSS/FreshRSS#7233] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I can't deploy FreshRSS on azure, because Mysql database required ssl, so I  get a error 

**Describe the solution you’d like**
Add a text field to copy the pem certificate content.

**Describe alternatives you’ve considered**
Maybe edit de config.php but I can't find any similar issue.

**Additional context**
`Blast! Verify your database configuration. : Access to database is denied for `user`: SQLSTATE[HY000] [3159] Connections using insecure transport are prohibited while --require_secure_transport=ON.`

**Comment by Alkarex** [comment]
Hello,
How did you install FreshRSS? Are you for instance using a Docker image? If yes, which?

**Comment by davisdre** [comment]
Interesting, I'm running into this same issue. 

I'm running Azure Web App (container setup) using `linuxserver/freshrss:latest`.

**Comment by Alkarex** [comment]
@davisdre Please try with `freshrss/freshrss:edge`
And what are the details of your MySQL server?
(You can also just use the build-in SQLite instead of MySQL)

**Comment by davisdre** [comment]
@Alkarex Thank you for the response.
I've changed to Docker image `freshrss/freshrss:edge` .
For MySQL server:
Azure Database for MySQL flexible server
Version: 8.0
Server Parameters, '[require_secure_transport](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_require_secure_transport)' = `On`

I believe for connectivity; I need to use `MYSQL_SSL_CA` to reference the provided certificate from the Azure Database for MySQL flexible server. That part is what I'm not familiar with how to implement the CA with FreshRSS to reference to it for MySQL connection. Any assistance is much appreciated.

**Comment by davisdre** [comment, marked as answer]
@Alkarex #868 resolved my issue. 

1. Deploy `freshrss-template.bicep`.
2. Upload .pem to freshrss-data folder.
3. Enable azure public services networking and change server parameters `requrie_secure_transport` to `Off` on the Azure Database for MySQL flexible server.
4. Complete initial FreshRSS setup.
5. Change server parameters `requrie_secure_transport` to `On`.
6. Modify the `config.php` in the freshrss-data folder that was generated after step 4. to reference the SSL CA.
    ```php
    'pdo_options' => array(
      PDO::MYSQL_ATTR_SSL_CA      => getenv('MYSQL_ATTR_SSL_CA'),
    ),
    ```
7. Restart Azure web app
8. Sign again. And success
